stdlib/public/core/StringCreate.swift

```diff
@@ -19,33 +19,54 @@ internal func _allASCII(_ input: UnsafeBufferPointer<UInt8>) -> Bool {
   //
   // TODO(String performance): SIMD-ize
   //
-  let ptr = input.baseAddress._unsafelyUnwrappedUnchecked
-  var i = 0
-
   let count = input.count
-  let stride = MemoryLayout<UInt>.stride
-  let address = Int(bitPattern: ptr)
-
-  let wordASCIIMask = UInt(truncatingIfNeeded: 0x8080_8080_8080_8080 as UInt64)
-  let byteASCIIMask = UInt8(truncatingIfNeeded: wordASCIIMask)
+  var ptr = UnsafeRawPointer(input.baseAddress._unsafelyUnwrappedUnchecked)
+  var i = 0
 
-  while (address &+ i) % stride != 0 && i < count {
-    guard ptr[i] & byteASCIIMask == 0 else { return false }
-    i &+= 1
+  let asciiMask64 = 0x8080_8080_8080_8080 as UInt64
+  let asciiMask32 = UInt32(truncatingIfNeeded: asciiMask64)
+  let asciiMask16 = UInt16(truncatingIfNeeded: asciiMask64)
+  let asciiMask8 = UInt8(truncatingIfNeeded: asciiMask64)
+  
+  let end128 = ptr + count & ~(MemoryLayout<(UInt64, UInt64)>.stride &- 1)
+  let end64 = ptr + count & ~(MemoryLayout<UInt64>.stride &- 1)
+  let end32 = ptr + count & ~(MemoryLayout<UInt32>.stride &- 1)
+  let end16 = ptr + count & ~(MemoryLayout<UInt16>.stride &- 1)
+  let end = ptr + count
+
+  
+  while ptr < end128 {
```

Comment:
We could go wider than 128 here, worth experimenting. My thinking with this was that a) most strings are relatively short, and b) this'll at least make sure we're using load-pair in the loop on arm64.

Reply:
Going to 256 wide hurts benchmark results locally, probably need to do that SIMD if/when we do it

Reply:
Also tried SIMD and it was dramatically (like ~2.5x) slower